catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/SourceOperations.java

```diff
@@ -264,27 +279,13 @@ boolean isSourceAvailable(Source source) {
       LOGGER.debug("source is null, therefore not available");
       return false;
     }
-    try {
-      if (LOGGER.isDebugEnabled()) {
-        LOGGER.debug("Checking if source \"{}\" is available...", source.getId());
-      }
-
-      // source is considered available unless we have checked and seen otherwise
-      boolean available = true;
-      Source cachedSource = frameworkProperties.getSourcePoller().getCachedSource(source);
-      if (cachedSource != null) {
-        available = cachedSource.isAvailable();
-      }
 
-      if (!available) {
-        LOGGER.info("source \"{}\" is not available", source.getId());
-      }
-      return available;
-    } catch (ServiceUnavailableException e) {
-      LOGGER.info("Caught ServiceUnavaiableException", e);
-      return false;
-    } catch (Exception e) {
-      LOGGER.info("Caught Exception", e);
+    // source is considered unavailable unless we have checked and seen otherwise
+    final Optional<Status> sourceStatusOptional = sourceStatusCache.getCachedValueForSource(source);
+    if (sourceStatusOptional.isPresent()) {
```

Comment:
❓How about this instead?
```suggestion
    return (boolean) sourceStatusOptional
        .map(s -> s == Status.AVAILABLE)
        .orElseGet(
          () -> {
            LOGGER.debug("Unknown source status for source id={}", source.getId());
            return false;
          });
```

Reply:
I don't want to log "unknown status" in the case that the last known status is unavailable.

Reply:
Based on the `Optional.map()` JavaDoc, I _think_ the `orElseGet()` will only be executed if `sourceStatusOptional` is empty. In other words, if `sourceStatusOptional` contains `Status.UNAVAILABLE`, `map()` will return an `Optional<Boolean>` containing `FALSE` and will not log anything. Is that what you're looking for?

Reply:
Oh, you're right 🤦‍♀️. Good suggestion. :+1: